gcp-secret-manager/src/main/java/io/micronaut/gcp/secretmanager/SecretManagerConfigurationClient.java

```diff
@@ -105,14 +108,15 @@ private Map<Integer, String> configCandidates(Environment environment) {
         int priority = EnvironmentPropertySource.POSITION + 150;
 
         if (configurationProperties.isDefaultConfigEnabled()) {
-            Optional<String> applicationName = environment.getProperty("micronaut.application.name", String.class);
-            candidates.put(EnvironmentPropertySource.POSITION + 101, "application");
-            applicationName.ifPresent(s -> candidates.put(EnvironmentPropertySource.POSITION + 102, s));
-
+            String applicationName = environment.getProperty(MICRONAUT_APPLICATION_NAME, String.class).orElse(null);
```

Comment:
@sdelamo I'd like to know what's your motivation to replace optionals with nullable references? Performance? Readability?

Reply:
I find this type of optimizations kind of premature, but I admit null checks will be somewhat faster in most scenarios.